webofneeds/won-owner-webapp/src/main/webapp/config/details/location.js

```diff
@@ -422,3 +209,171 @@ export const travelAction = {
+          "@type": "s:GeoCoordinates",
+          "s:latitude": seCorner.lat.toFixed(6),
+          "s:longitude": seCorner.lng.toFixed(6),
+        },
+      };
+}
+
+function parseSPlace(jsonldLocation) {
+  // const jsonldLocation = jsonLDImm && jsonLDImm.get("won:hasLocation");
+  if (!jsonldLocation) return undefined; // NO LOCATION PRESENT
+
+  const location = parsePlaceLeniently(jsonldLocation);
+
+  if (
+    location &&
+    location.address &&
+    location.lat &&
+    location.lng &&
+    location.nwCorner.lat &&
+    location.nwCorner.lng &&
+    location.seCorner.lat &&
+    location.seCorner.lng
+  ) {
+    return Immutable.fromJS(location);
+  } else {
+    console.error(
+      "Cant parse location, data is an invalid location-object: ",
+      jsonldLocation
+    );
+    return undefined;
+  }
+}
+
+/**
+ * Parses json-ld of an `s:Place` in a best-effort kinda style.
+ * Any data missing in the RDF will be missing in the result object.
+ * @param {*} jsonldLocation
+ */
+function parsePlaceLeniently(jsonldLocation) {
+  if (!jsonldLocation) return undefined; // NO LOCATION PRESENT
+
+  const jsonldLocationImm = Immutable.fromJS(jsonldLocation);
+
+  const parseFloatFromLocation = path =>
+    won.parseFrom(jsonldLocationImm, path, "xsd:float");
+
+  const place = {
+    address: won.parseFrom(jsonldLocationImm, ["s:name"], "xsd:string"),
+    lat: parseFloatFromLocation(["s:geo", "s:latitude"]),
+    lng: parseFloatFromLocation(["s:geo", "s:longitude"]),
+    // nwCorner if present (see below)
+    // seCorner if present (see below)
+  };
+
+  const nwCornerLat = parseFloatFromLocation([
+    "won:hasBoundingBox",
+    "won:hasNorthWestCorner",
+    "s:latitude",
+  ]);
+  const nwCornerLng = parseFloatFromLocation([
+    "won:hasBoundingBox",
+    "won:hasNorthWestCorner",
+    "s:longitude",
+  ]);
+  const seCornerLat = parseFloatFromLocation([
+    "won:hasBoundingBox",
+    "won:hasSouthEastCorner",
+    "s:latitude",
+  ]);
+  const seCornerLng = parseFloatFromLocation([
+    "won:hasBoundingBox",
+    "won:hasSouthEastCorner",
+    "s:longitude",
+  ]);
+
+  if (nwCornerLat || nwCornerLng) {
+    place.nwCorner = {
+      lat: nwCornerLat,
+      lng: nwCornerLng,
+    };
+  }
+  if (seCornerLat || seCornerLng) {
+    place.seCorner = {
+      lat: seCornerLat,
+      lng: seCornerLng,
+    };
+  }
+  return place;
+}
+
+function sPlaceToHumanReadable({ value, includeLabel }) {
+  if (value) {
+    let humanReadable;
+    if (value.name) {
+      humanReadable = value.name;
+    } else if (value.address) {
+      humanReadable = value.address;
+    } else {
+      const locationLat = value.lat && value.lat.toFixed(6);
+      const locationLng = value.lng && value.lng.toFixed(6);
+      if (locationLat && locationLng) {
+        humanReadable = "@(" + locationLat + " , " + locationLng + ")";
+      }
+    }
+    if (humanReadable) {
+      return includeLabel
+        ? this.label + ": " + humanReadable.trim()
```

Comment:
This use of `this.label` causes the error documented by @maxstolze - `this` is expected to be the detail definition, which is no longer true due to the refactoring.